File: plugin/main/java/nl/utwente/ewi/fmt/uppaalSMC/urpal/properties/BreadthFirstBaseLine.java
package nl.utwente.ewi.fmt.uppaalSMC.urpal.properties;

import java.io.IOException;
import java.util.function.Consumer;

import com.uppaal.engine.EngineException;
import com.uppaal.model.core2.Document;
import com.uppaal.model.system.UppaalSystem;

import nl.utwente.ewi.fmt.uppaalSMC.NSTA;

@SanityCheck(name = "Breadth first")
public class BreadthFirstBaseLine extends AbstractProperty {

    @Override
    public void doCheck(NSTA nsta, Document doc, UppaalSystem sys, Consumer<SanityCheckResult> cb) {
        long startTime = System.currentTimeMillis();
        try {
            engineQuery(sys, "A[](true)", DEFAULT_OPTIONS_BFS, (gf, ts) -> {
                System.out.println("time milis: " + (System.currentTimeMillis() - startTime));
                System.out.println("max mem: " + (maxMem));
            });
        } catch (IOException | EngineException e) {
            e.printStackTrace();
        }
    }

}

File: plugin/main/java/nl/utwente/ewi/fmt/uppaalSMC/urpal/properties/BreadthFirstBaseLine.kt
package nl.utwente.ewi.fmt.uppaalSMC.urpal.properties

import com.uppaal.model.core2.Document
import com.uppaal.model.system.UppaalSystem

import nl.utwente.ewi.fmt.uppaalSMC.NSTA

@SanityCheck(name = "Breadth first")
class BreadthFirstBaseLine : AbstractProperty() {

    override fun doCheck(nsta: NSTA, doc: Document, sys: UppaalSystem, cb: (SanityCheckResult) -> Unit) {
        val startTime = System.currentTimeMillis()

        AbstractProperty.engineQuery(sys, "A[](true)", AbstractProperty.DEFAULT_OPTIONS_BFS) { _, _ ->
            println("time milis: ${System.currentTimeMillis() - startTime}")
            println("max mem: ${AbstractProperty.maxMem}")
        }


    }

}
